pf/internal/plugin/provider_server.go

```diff
@@ -55,7 +56,6 @@ func (p *providerServer) marshalOptions(label string) pl.MarshalOptions {
 	return pl.MarshalOptions{
 		Label:         label,
 		KeepUnknowns:  true,
-		KeepSecrets:   p.keepSecrets,
```

Comment:
This change feels like it should break some tests somewhere. We must be able to return secrets to the engine when schema mandates secrets. I'd like to dive in why we're lacking test coverage here.

Would https://github.com/pulumi/pulumi-terraform-bridge/pull/1050  break perhaps?

Reply:
Yes. I'm reverting this part.